File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        StringBuilder sb=new StringBuilder();

        int index=0;

        for(String str:s.split("")){
            if(!str.equals("*")){
                sb.append(str);
                index++;
            }
            else{
                sb.deleteCharAt(--index);
            }
        }

        return sb.toString();
    }
}

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.c
char* removeStars(char* s) {
    int read=0,write=0;
    while(s[read]){
        if(s[read]!='*'){
            s[write++]=s[read];
        }
        else{
            write--;
        }
        read++;
    }
    s[write]='\0';
    
    return s;
}